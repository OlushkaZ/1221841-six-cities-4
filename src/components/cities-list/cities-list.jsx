import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../redux/actions/actions.js";
import {City} from "../../constants/page.js";
import {getActiveCity} from "../../reducer/page/selectors.js";

const cities = Object.values(City);

function CitiesList({city: activeCity, onChooseCityClick}) {
  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">

          {cities.map((city) => {
            const onClick = () => onChooseCityClick(city);
            return (
              <li
                className="locations__item"
                key={city}
                onClick={onClick}>
                <a
                  className={`locations__item-link tabs__item  ${city === activeCity ? `tabs__item--active` : ``}`}
                  href="#">
                  <span>{city}</span>
                </a>
              </li>
            );
          })}
        </ul>
      </section>
    </div>
  );
}

CitiesList.propTypes = {
  city: PropTypes.string.isRequired,
  onChooseCityClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  city: getActiveCity(state),
});

const mapDispatchToProps = (dispatch) => ({
  onChooseCityClick(city) {
    dispatch(ActionCreator.chooseCity(city));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
export {CitiesList};


