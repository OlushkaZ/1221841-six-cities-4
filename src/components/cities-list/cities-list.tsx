import * as React from "react";
import {connect} from "react-redux";
import {PureComponent} from "react";
import {ActionCreator} from "../../redux/page/page";
import {getActiveCity} from "../../redux/page/selectors";
import {getCities} from "../../redux/offers-data/selectors";

interface Props {
  city: string;
  cities: Array<string>;
  setDefaultCity: (city: string) => void;
  onChooseCityClick: (city: string) => void;
}


class CitiesList extends React.PureComponent<Props, {}> {
  componentDidMount() {
    const {city: activeCity, cities, setDefaultCity} = this.props;
    return activeCity === null && setDefaultCity(cities[0]);
  }

  render() {
    const {city: activeCity, onChooseCityClick, cities} = this.props;
    return (
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((city) => {
              const handleChooseCityClick = () => onChooseCityClick(city);
              return (
                <li
                  className="locations__item"
                  key={city}
                  onClick={handleChooseCityClick}>
                  <a
                    className={`locations__item-link tabs__item  ${city === activeCity ? `tabs__item--active` : ``}`}
                    href="#">
                    <span>{city}</span>
                  </a>
                </li>
              );
            })}
          </ul>
        </section>
      </div>
    );
  }
}


const mapStateToProps = (state) => ({
  city: getActiveCity(state),
  cities: getCities(state),
});

export const mapDispatchToProps = (dispatch) => ({
  onChooseCityClick(city) {
    dispatch(ActionCreator.chooseCity(city));
  },

  setDefaultCity(city) {
    dispatch(ActionCreator.chooseCity(city));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
export {CitiesList};


