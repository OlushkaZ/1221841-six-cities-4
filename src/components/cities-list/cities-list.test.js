import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list.jsx";

const props = {
  cities: [`Paris`, `Moscow`, `Amsterdam`, `Honkong`],
  city: `Moscow`,
  onChooseCityClick: jest.fn(),
};

it(`CitiesList render`, () => {
  const tree = renderer
  .create(
      <CitiesList {...props}/>
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
