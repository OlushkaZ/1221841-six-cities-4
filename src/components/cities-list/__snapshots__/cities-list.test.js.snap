// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CotiesList render 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item  "
          href="#"
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item  tabs__item--active"
          href="#"
        >
          <span>
            Moscow
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item  "
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
        onClick={[Function]}
      >
        <a
          className="locations__item-link tabs__item  "
          href="#"
        >
          <span>
            Honkong
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
